### DISCUSSION TITLE
Disappearance of the wingpanel

### DISCUSSION THREAD

**Discussion by TheSirC** [elementary/wingpanel#622] [Q&A]
Hello !
After a simple `apt-get update -y && sudo apt-get upgrade -y && sudo apt-get dist-upgrade -y` my wingpanel disappeared. Totally. 

When I call it via the console I get this [output](https://github.com/elementary/wingpanel/files/1786815/err.log)

That's really crippling :cry: 

You are doing a great job guys ! I have been using elementary since the beginning ! Thanks in advance.

**Comment by jeremypw** [comment]
I had a similar problem on Juno unstable which I fixed by reinstalling switchboard and all its plugs from source but unfortunately I did not locate the cause.

Correction:  All the plugs for which I had installed a PR version at some point (a fair number), not all.

**Comment by TheSirC** [comment]
Interesting do you remember which one exactly ?

**Comment by jeremypw** [comment]
I have just had this problem reoccur and I did not install any switchboard plug PRs so I am not sure whether that was the cause or not now.  It occurred after the window manager stopped responding to clicks on the "close window" button in the middle of a session.  I had to shutdown from the terminal.  On re-booting, wingpanel failed to appear.  On rebooting again the wingpanel is back so I am puzzled.

**Comment by jeremypw** [comment]
Another occurrence - this seemed to be associated with memory problems - other apps were becoming slow although no high CPU usage.  Killing wingpanel process a couple of times brought back a responsive panel but only for a short time.  The system soon locked up completely.

**Comment by jeremypw** [comment]
Out of interest, do you have "auto-login" enabled?  It maybe that wingpanel expects access to the session key-ring.

**Comment by TheSirC** [comment]
Not that I know of.
Same for CPU. By just "undoing" my update via Timeshift, I am able to come back to a stable wingpanel... So I really think the update is linked to it.

**Comment by leshik** [comment]
What's the status of this issue? ElementaryOS is completely unusable now. Thinking about switching to EndlessOS :)

**Comment by donadigo** [comment]
What are you running? Daily or stable? There were a couple of changes that could cause this to happen. Specifically the switch to RDNN scheme and changing when the content of the indicators loads...
I also do not think that any of those changes got released to Loki stable at all.

The only thing I can suggest you is to revert to previous version because it's unclear to me what change caused this, it doesn't even have to be the issue in the wingpanel itself.

**Comment by leshik** [comment]
This is Loki/stable install on my mom’s computer. It had the problem with wingpanel during the whole year that she has it installed, the panel freezed time to time, and crashed often. I tried I think every advice that I’ve read here, including deinstalling ALL indicators & also rebuilding from source, but it didn’t help. Then I hadn’t updated the computer for a while (since February or March I believe), and after I did update it last week the panel completely disappeared and doesn’t work anymore. It can come up after 5 min after the boot, but usually doesn’t. I can’t restore previous versions as there are no old packages in the repository. From what I can see wingpanel wasn’t updated since the end of 2017 in stable, so I conclude the situation became worse after some of its dependencies has updated (no idea which one exactly).
